Sources/Core/Views/TextField/TextField.swift

```diff
@@ -0,0 +1,304 @@
+//
+//  File.swift
+//  
+//
+//  Created by Arlind Dushi on 01/20/2022.
+//
+
+import UIKit
+
+public protocol RuleValidatable {
+  var error: String { get }
+  func validate(_ input: String?) -> Bool
+}
+
+public class TextField: UIView {
+  private let stackView = UIStackView()
+  private let titleLabel = PaddingLabel()
+  private let valueContainerView = UIView()
+  private let valueTextField = UITextField()
+  private let errorLabel = UILabel()
+  private var rule: RuleValidatable?
+  private var textFieldHeightAnchor: NSLayoutConstraint?
+
+  public var onTextChange: ((String) -> Void)?
+  public var onReturnKeyPressed: (() -> Void)?
+  public var onDidBegindEditing: (() -> Void)?
+  public var onDidEndEditing: (() -> Void)?
+
+  public init(with rule: RuleValidatable? = .none) {
+    super.init(frame: .zero)
+    self.rule = rule
+    setupViews()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  @discardableResult
+  override public func becomeFirstResponder() -> Bool {
+    valueTextField.becomeFirstResponder()
+  }
+  
+  @discardableResult
+  override public func resignFirstResponder() -> Bool {
+    valueTextField.resignFirstResponder()
+  }
+  
+  override public var isFirstResponder: Bool {
+    valueTextField.isFirstResponder
+  }
+  
+  public var inputFieldHeight: CGFloat = 44 {
+    didSet { textFieldHeightAnchor?.constant = inputFieldHeight }
+  }
+  
+  public var inputFieldCornerRadius: CGFloat = 10 {
+    didSet { valueContainerView.layer.cornerRadius = inputFieldCornerRadius }
+  }
+  
+  private var error: String? {
+    didSet { errorStateChanged() }
+  }
+  
+  public var hideKeyboardOnReturn = false
+  public var shouldChangeCharacters: Bool = true
```

Comment:
Could you add more context as to what this bool does, and also maybe provide a way to add text validation for `textField(_:shouldChangeCharactersIn:replacementString:)`

Reply:
After rethinking this I think offering a protocol to the implementer to do its own implementation is a better way to go. Will do the necessary changes here.